core/modules/layout/includes/layout.class.inc

```diff
@@ -262,13 +262,39 @@ class Layout {
       $function($this->name, $data);
     }
 
+    // Save a new configuration file if the layout has been renamed.
     if (isset($this->original_name) && $this->original_name != $this->name) {
       config('layout.layout.' . $this->original_name)->delete();
     }
+
+    // Save the layout configuration file.
     config('layout.layout.' . $this->name)
       ->setData($data)
       ->save();
 
+    // Parse each BlockText block for file IDs.
+    $fids = array();
+    foreach ($this->content as $content) {
+      if (is_a($content, 'BlockText')) {
+        $block_content = $content->settings['content'];
+        $block_fids = filter_parse_file_fids($block_content);
+        if (isset($block_fids)) {
+          $fids = array_merge($fids, $block_fids);
+        }
+      }
+    }
+    // Save each file as permanent, preventing it from being deleted.
+    $files = file_load_multiple($fids);
+    foreach ($files as $fid => $file) {
+      if ($file && $file->status !== FILE_STATUS_PERMANENT) {
+        $file->status = FILE_STATUS_PERMANENT;
+        $file->save();
+      }
+      // Essentially this claims that the file is "self-referencing", which means
+      // it will never be deleted until a file management tool is available.
+      file_usage_add($file, 'file', 'file', $file->fid);
+    }
+
```

Comment:
Isn't this duplication here? These blocks can't really be used except in a layout, which means on layout save, this process will happen twice.